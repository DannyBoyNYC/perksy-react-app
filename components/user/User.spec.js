const React = require('react');
const { shallow } = require('enzyme');

const User = require('./index').default;

describe('<User.* />', () => {
  let component;
  const requiredProps = {
    brandName: 'Perksy',
    firstName: 'Jane',
    lastName: 'Doe'
  };

  describe('<User.Avatar />', () => {
    beforeEach(() => {
      component = shallow(<User.Avatar {...requiredProps} />).dive();
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should have the correct BEM class names', () => {
      expect(component).toHaveClassName('user user--avatar');
    });
  });
});
