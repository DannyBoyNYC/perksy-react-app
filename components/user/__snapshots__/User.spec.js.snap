// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User.* /> <User.Avatar /> should render correctly with all required props 1`] = `
<div
  className="user user--avatar"
>
  <div
    className="user__image"
  />
  <div
    className="user__name"
  >
    <span
      className="user__first-name"
    >
      Jane
    </span>
    <span
      className="user__last-name"
    >
      Doe
    </span>
  </div>
  <div
    className="user__brand"
  >
    Perksy
  </div>
</div>
`;
